src/orders/src/main/java/com/amazon/sample/orders/metrics/OrdersMetrics.java

```diff
@@ -19,22 +19,70 @@
 package com.amazon.sample.orders.metrics;
 
 import com.amazon.sample.events.orders.OrderCreatedEvent;
+import com.amazon.sample.orders.entities.OrderItemEntity;
 import io.micrometer.core.instrument.Counter;
 import io.micrometer.core.instrument.MeterRegistry;
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.event.TransactionalEventListener;
 
+import java.util.concurrent.atomic.AtomicInteger;
+
 @Component
 public class OrdersMetrics {
 
     private Counter orderCreatedCounter;
+    private MeterRegistry meterRegistry;
+    private Counter pocketWatchCounter;
+    private Counter woodWatchCounter;
+    private Counter gentlemanWatchCounter;
+    private Counter classicWatchCounter;
+
+    public static final String POCKET_WATCH = "Pocket Watch";
+    public static final String GENTLEMAN_WATCH = "Gentleman";
+    public static final String CHRONOGRAF_WATCH = "Chronograf Classic";
+    public static final String WOOD_WATCH = "Wood Watch";
 
     public OrdersMetrics(MeterRegistry meterRegistry) {
+        this.meterRegistry = meterRegistry;
         this.orderCreatedCounter = meterRegistry.counter("orders_created");
+        this.pocketWatchCounter = Counter.builder("watch.orders")
```

Comment:
Would it be practical to create a Map of Counter objects using the productId as the key? This would be a bit more flexible and less "hard-coded". We could change the logic in onOrderCreated to check if a counter exists for a given productid and if not then create a new one.

Reply:
Move the type to pull from map using productId. good call out